src/keys.ts

```diff
@@ -50,7 +50,9 @@ type IndicesAfterSpread<
       : Iterations["length"];
 
 type ObjectKeys<T> =
-  T extends Record<PropertyKey, never> ? [] : Array<EnumerableStringKeyOf<T>>;
+  T extends Record<PropertyKey, never>
+    ? []
+    : Array<EnumerableStringKeyOf<Required<T>>>;
```

Comment:
Shouldn't the `Required` wrapper part be part of the `EnumerableStringKeyOf` type and not here?

Reply:
You are right thank you, I will integrate it in `EnumerableStringKeyOf`.